prog/mutation.go

```diff
@@ -374,6 +374,30 @@ func (t *BufferType) mutate(r *randGen, s *state, arg Arg, ctx ArgCtx) (calls []
 	case BufferText:
 		data := append([]byte{}, a.Data()...)
 		a.data = r.mutateText(t.Text, data)
+	case BufferCompressed:
```

Comment:
You mentioned that BufferCompressed is not mutated with hints, but I don't see any changes in generateHints. It only ignores BufferFilename/Strings, but BufferCompressed should be mutated.
I think we need to skip BufferCompressed in generateHints and add a comment to reconsider it in future. We could decompress, apply hints and compress again. On one hand I think hints mutation can be useful for images, but on the other hand it may produce way too many mutants with the default handling we have for buffers, so I think it's better to skip BufferCompressed for now.